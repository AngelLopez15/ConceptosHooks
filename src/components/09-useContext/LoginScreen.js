import React, { useContext } from 'react'
import { UserContext } from './UserContext'

export const LoginScreen = () => {
    
    // Obteniendo la referencia al userContext
    const {user,setUser} = useContext(UserContext)

    return (
        <div>
            <h1>LoginScreen</h1>
            <hr />
            <pre>
                {JSON.stringify(user, null, 3)}
            </pre>
            <button className="btn btn-primary"
                onClick={()=>setUser({
                    id:123,
                    name:'Angel'
                })}
            >
                Login
            </button>
        </div>
    )
}
