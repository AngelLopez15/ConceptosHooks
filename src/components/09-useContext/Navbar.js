import React from 'react'
import { Link, NavLink } from 'react-router-dom'

export const Navbar = () => {
    // Navlink solo nos sirve para poner una clase y se muestre activado la ruta donde estamos
    return (
        <nav className="navbar navbar-expand-lg navbar-dark bg-dark">
            <Link className="navbar-brand" to="/">useContext</Link>
            <div className="navbar-nav">
                <NavLink exact activeClassName="active" className="nav-item nav-link" to="/">Home</NavLink>
                <NavLink exact activeClassName="active" className="nav-item nav-link" to="/login">Login</NavLink>
                <NavLink exact activeClassName="active" className="nav-item nav-link" to="/about">About</NavLink>
            </div>
        </nav>
    )
}
