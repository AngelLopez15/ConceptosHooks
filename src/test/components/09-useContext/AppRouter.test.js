import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { configure, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { AppRouter } from '../../../components/09-useContext/AppRouter';
import { UserContext } from '../../../components/09-useContext/UserContext';

describe('Pruebas en <AppRouter />', () => {
  
  configure({adapter: new Adapter()});

  const user={
    id:1,
    name:'Angel'
  }

  const wrapper = mount(
    <UserContext.Provider
      value={{user}}
    >
      <AppRouter />
    </UserContext.Provider>
  )

  test('Debe de mostrarse correctamente', () => {
    
    expect(toJson(wrapper)).toMatchSnapshot();

  })
  

});