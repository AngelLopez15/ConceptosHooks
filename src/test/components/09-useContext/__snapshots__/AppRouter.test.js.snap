// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AppRouter /> Debe de mostrarse correctamente 1`] = `
<AppRouter>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <div>
        <Navbar>
          <nav
            className="navbar navbar-expand-lg navbar-dark bg-dark"
          >
            <Link
              className="navbar-brand"
              to="/"
            >
              <LinkAnchor
                className="navbar-brand"
                href="/"
                navigate={[Function]}
              >
                <a
                  className="navbar-brand"
                  href="/"
                  onClick={[Function]}
                >
                  useContext
                </a>
              </LinkAnchor>
            </Link>
            <div
              className="navbar-nav"
            >
              <NavLink
                activeClassName="active"
                className="nav-item nav-link"
                exact={true}
                to="/"
              >
                <Link
                  aria-current="page"
                  className="nav-item nav-link active"
                  style={Object {}}
                  to={
                    Object {
                      "hash": "",
                      "pathname": "/",
                      "search": "",
                      "state": null,
                    }
                  }
                >
                  <LinkAnchor
                    aria-current="page"
                    className="nav-item nav-link active"
                    href="/"
                    navigate={[Function]}
                    style={Object {}}
                  >
                    <a
                      aria-current="page"
                      className="nav-item nav-link active"
                      href="/"
                      onClick={[Function]}
                      style={Object {}}
                    >
                      Home
                    </a>
                  </LinkAnchor>
                </Link>
              </NavLink>
              <NavLink
                activeClassName="active"
                className="nav-item nav-link"
                exact={true}
                to="/login"
              >
                <Link
                  aria-current={null}
                  className="nav-item nav-link"
                  to={
                    Object {
                      "hash": "",
                      "pathname": "/login",
                      "search": "",
                      "state": null,
                    }
                  }
                >
                  <LinkAnchor
                    aria-current={null}
                    className="nav-item nav-link"
                    href="/login"
                    navigate={[Function]}
                  >
                    <a
                      aria-current={null}
                      className="nav-item nav-link"
                      href="/login"
                      onClick={[Function]}
                    >
                      Login
                    </a>
                  </LinkAnchor>
                </Link>
              </NavLink>
              <NavLink
                activeClassName="active"
                className="nav-item nav-link"
                exact={true}
                to="/about"
              >
                <Link
                  aria-current={null}
                  className="nav-item nav-link"
                  to={
                    Object {
                      "hash": "",
                      "pathname": "/about",
                      "search": "",
                      "state": null,
                    }
                  }
                >
                  <LinkAnchor
                    aria-current={null}
                    className="nav-item nav-link"
                    href="/about"
                    navigate={[Function]}
                  >
                    <a
                      aria-current={null}
                      className="nav-item nav-link"
                      href="/about"
                      onClick={[Function]}
                    >
                      About
                    </a>
                  </LinkAnchor>
                </Link>
              </NavLink>
            </div>
          </nav>
        </Navbar>
        <div
          className="container"
        >
          <Switch>
            <Route
              component={[Function]}
              computedMatch={
                Object {
                  "isExact": true,
                  "params": Object {},
                  "path": "/",
                  "url": "/",
                }
              }
              exact={true}
              location={
                Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                }
              }
              path="/"
            >
              <HomeScreen
                history={
                  Object {
                    "action": "POP",
                    "block": [Function],
                    "createHref": [Function],
                    "go": [Function],
                    "goBack": [Function],
                    "goForward": [Function],
                    "length": 1,
                    "listen": [Function],
                    "location": Object {
                      "hash": "",
                      "pathname": "/",
                      "search": "",
                      "state": undefined,
                    },
                    "push": [Function],
                    "replace": [Function],
                  }
                }
                location={
                  Object {
                    "hash": "",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  }
                }
                match={
                  Object {
                    "isExact": true,
                    "params": Object {},
                    "path": "/",
                    "url": "/",
                  }
                }
              >
                <div>
                  <h1>
                    HomeScreen
                  </h1>
                  <hr />
                  <pre>
                    {
   "id": 1,
   "name": "Angel"
}
                  </pre>
                </div>
              </HomeScreen>
            </Route>
          </Switch>
        </div>
      </div>
    </Router>
  </BrowserRouter>
</AppRouter>
`;
